addons/knobs/src/registerKnobs.js

```diff
@@ -15,12 +15,16 @@ function setPaneKnobs(timestamp = +new Date()) {
 }
 
 function knobChanged(change) {
-  const { name, value } = change;
+  const { name, type, value } = change;
 
   // Update the related knob and it's value.
   const knobOptions = knobStore.get(name);
 
-  knobOptions.value = value;
+  if (type === 'boolean') {
+    knobOptions.value = !knobOptions.value;
```

Comment:
what if the default value is `false` and switched to `true` but since `value` is `undefined`, this will set `knobOptions.value` as `false` which is the wrong value. 

I would suggest a `undefined` check.

Reply:
> initial toggled to `false`

What do you mean exactly? That the default value is true and the user toggles it to false? That the default is true and the user toggled it to false, and that toggle persisted after refresh? I'm kinda confused so could you elaborate more?

Reply:
@wuweiweiwu in my change, I'm using the `knopOptions.value` stored in the current knob to calculate the new `knobOptions.value`. So i'm not using the `value` destructured from the incoming `change` object. That value is always `undefined`, so it's impossible to defer the correct boolean from that value. The knob itself initializes its `knobOptions.value` with the setted default value, so my change is basically proposing to flip that value every time, instead of relying on the incoming event value. Does this solve your issue? Sorry for the confusion.

Reply:
> That value is always `undefined`

I would prefer to fix this part

Reply:
I will see what I can do

Reply:
@Keraito ah my bad. I thought you were doing `!value` instead.